yt_dlp/extractor/rokfin.py

```diff
@@ -0,0 +1,561 @@
+    _CHANNEL_BASE_URL = 'https://rokfin.com/'
+
+
+class RokfinPostIE(RokfinSingleVideoIE):
+    IE_NAME = 'rokfin:post'
+    _VALID_URL = r'https?://(?:www\.)?rokfin\.com/(?P<id>post/[^/]+)'
+    _TESTS = [{
+        'url': 'https://www.rokfin.com/post/57548/Mitt-Romneys-Crazy-Solution-To-Climate-Change',
+        'info_dict': {
+            'id': 'post/57548',
+            'ext': 'mp4',
+            'title': 'Mitt Romney\'s Crazy Solution To Climate Change',
+            'thumbnail': 're:https://img.production.rokfin.com/.+',
+            'upload_date': '20211023',
+            'timestamp': 1634998029,
+            'creator': 'Jimmy Dore',
+            'channel': 'Jimmy Dore',
+            'channel_id': 65429,
+            'channel_url': 'https://rokfin.com/TheJimmyDoreShow',
+            'duration': 213.0,
+            'availability': 'public',
+            'live_status': 'not_live'
+        }
+    }]
+
+    def _real_extract(self, url_from_user):
+        if self.get_param('username') or self.get_param('password'):
+            self.report_warning('site logins are unsupported')
+
+        video_id = self._match_id(url_from_user)
+        downloaded_json_meta_data = self._download_json(self._META_DATA_BASE_URL + video_id, video_id, fatal=False) or {}
+        video_formats_url = url_or_none(traverse_obj(downloaded_json_meta_data, ('content', 'contentUrl')))
+        availability = self._availability(
+            is_private=False,
+            needs_premium=True if downloaded_json_meta_data.get('premiumPlan') == 1 else False if downloaded_json_meta_data.get('premiumPlan') == 0 else None,
+            # premiumPlan = 0 - no-premium content
+            # premiumPlan = 1 - premium-only content
+            needs_subscription=False, needs_auth=False, is_unlisted=False)
+
+        if downloaded_json_meta_data.get('premiumPlan') not in (0, 1, None):
+            self.report_warning(f'unknown availability code: {downloaded_json_meta_data.get("premiumPlan")}. Rokfin extractor should be updated')
+
+        if video_formats_url:
+            if try_get(video_formats_url, lambda x: urlparse(x).path.endswith('.m3u8')):
+                frmts, subs = self._extract_m3u8_formats_and_subtitles(m3u8_url=video_formats_url, video_id=video_id, fatal=False)
+            else:
+                frmts = [{'url': video_formats_url}]
+                subs = None
+        else:
+            frmts = None
+            subs = None
+
+        if not frmts:
+            if availability == 'premium_only':
+                self.raise_no_formats(msg='premium content', video_id=video_id, expected=True)
+            elif video_formats_url:
+                self.raise_no_formats(msg='unable to download: missing meta data', video_id=video_id, expected=True)
+            else:
+                # We don't know why there is no (valid) meta data present.
+                self.raise_no_formats(msg='unable to download', video_id=video_id, expected=True)
```

Comment:
```suggestion
        if not frmts and availability == 'premium_only':
            self.raise_login_required(msg='This video is only available for premium users', True, method='cookies')
```

Reply:
1. Changed the first occurrence of `raise_no_formats` to `raise_login_required` (Also in RokfinStreamIE).
2. Still need the second and third `raise_no_formats` where they are right now. The second one happens when a valid video URL does not hold video formats. The third is when there is no valid video URL to begin with.
3. Your error message "The video is only ..." is misleading since the "video" may actually be a so-called "stack" which is a collection of content.
4. Changing the if-statement as you suggest will make it more redundant, as in

```
if not frmts and availability == 'premium_only':
    something
elif not frmts and video_formats_url:
    something
elif not frmts:
    something
```

Reply:
Can you provide examples links for 2,3?

Reply:
Not immediately. But the Murphy's law still applies.

Reply:
In other words, cases 2 and 3 are logically possible even in the absence of a concrete example.